From 8e931ebddc820180e2cc8c161142772cdf6240d3 Mon Sep 17 00:00:00 2001
From: Joel Selvaraj <joelselvaraj.oss@gmail.com>
Date: Sat, 10 Sep 2022 21:38:41 -0500
Subject: [PATCH 063/211] stalled! arm64: dts: qcom: sdm845-xiaomi-beryllium:
 add separate dai link for headphone mic

We can use existing multimedia+slimbus dai link for headphones mic too.
But it doesn't work well with the linux userspace side.
It wouldn't seamlessly switch between both the mics.
Adding a separate multimedia and slimbus dai fixes this issue.
---
 .../qcom/sdm845-xiaomi-beryllium-common.dtsi  | 28 ++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
index f8b3e30b904c..40fbca97a1bc 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
@@ -461,6 +461,10 @@ dai@1 {
 	dai@2 {
 		reg = <MSM_FRONTEND_DAI_MULTIMEDIA3>;
 	};
+
+	dai@3 {
+		reg = <MSM_FRONTEND_DAI_MULTIMEDIA4>;
+	};
 };
 
 &qupv3_id_0 {
@@ -523,6 +527,13 @@ cpu {
 		};
 	};
 
+	mm4-dai-link {
+		link-name = "MultiMedia4";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA4>;
+		};
+	};
+
 	dai-link-voicemmode1 {
 		link-name = "VoiceMMode1";
 		cpu {
@@ -562,7 +573,7 @@ codec {
 	};
 
 	slimcap-dai-link {
-		link-name = "SLIM Capture";
+		link-name = "SLIM WCD Capture";
 		cpu {
 			sound-dai = <&q6afedai SLIMBUS_0_TX>;
 		};
@@ -575,6 +586,21 @@ codec {
 			sound-dai = <&wcd9340 1>;
 		};
 	};
+
+	slimcap2-dai-link {
+		link-name = "SLIM WCD Capture 2";
+		cpu {
+			sound-dai = <&q6afedai SLIMBUS_1_TX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&wcd9340 3>; /* AIF2_CAP */
+		};
+	};
 };
 
 &tlmm {
-- 
2.55.0

